code/modules/mob/living/living.dm

```diff
@@ -1130,21 +1130,27 @@
 		return TRUE
 	return FALSE
 
-/mob/living/proc/ExtinguishMob()
-	if(on_fire)
+/**
+ * Extinguish all fire on the mob
+ *
+ * This removes all fire stacks, fire effects, alerts, and moods
+ * Signals the extinguishing.
+ */
+/mob/living/proc/extinguish_mob()
+	fire_stacks = 0
+	for(var/obj/effect/dummy/lighting_obj/moblight/fire/F in src)
+		qdel(F)
+	clear_alert("fire")
+	SEND_SIGNAL(src, COMSIG_CLEAR_MOOD_EVENT, "on_fire")
+	if(on_fire) //Only signal if mob was on fire
 		on_fire = FALSE
-		fire_stacks = 0
-		for(var/obj/effect/dummy/lighting_obj/moblight/fire/F in src)
-			qdel(F)
-		clear_alert("fire")
-		SEND_SIGNAL(src, COMSIG_CLEAR_MOOD_EVENT, "on_fire")
 		SEND_SIGNAL(src, COMSIG_LIVING_EXTINGUISHED, src)
-		update_fire()
+	update_fire()
 
 /mob/living/proc/adjust_fire_stacks(add_fire_stacks) //Adjusting the amount of fire_stacks we have on person
 	fire_stacks = clamp(fire_stacks + add_fire_stacks, -20, 20)
-	if(on_fire && fire_stacks <= 0)
-		ExtinguishMob()
+	if(fire_stacks <= 0)
+		extinguish_mob()
```

Comment:
Any reason to remove the `on_fire` check here?

Reply:
If we called change fire stacks and we now have less fire stacks then the minimum to be on fire then lets extinguish the mob clearing fire sprites and alerts.
This prevents mobs from being on fire visually having no fire stacks, and on_fire is false being unable to remove the fire effect.

Reply:
This is hiding a bug, however, because when `on_fire` is set to `FALSE`, the fire sprites and alerts are removed.
Which means that they are being added somewhere with it set as `FALSE`.

Reply:
I am moving this check to the extinguish_mob as an early return in next update.